#17 shift plugins (#37)
Inbibo/shift_docs: docs/reference/plugins.md

# Plugins

Shift supports the integration of custom Plugins. Shift Plugins are Qt based UIs which can interact with Shift to display additional information, perform actions or modify the behaviour of the application.

## Shift Native Plugins

> Shift ships with two example plugins which can be found in the *Tools* section in the top menu bar. The desired plugin can be opened by clicking on the corresponding entry.

Reviewer: I woudln't say they are examples, I see them more as native addons.
```suggestion
Shift ships with two plugins which can be found in the *Tools* section in the top menu bar. The desired plugin can be opened by clicking on the corresponding entry.

```

> Once a Plugin is opened, the Shift UI can be customized with the best suited layout.

Reviewer: Shouldn't we mention this in the description of plugins, instead of the Shift Native Plugins section?
Basically, as a general rule for all, Plugins are added as dockable widgers in the ShiftUI, meaning they can be resized and placed as desired.

Author: moved this part above as you suggested

> The *Operators List* Plugin helps us visualize the list of operators in the workflow and shows the their current progress.

Reviewer: I would also mention that shows the order of execution even before it gets executed.

Reviewer: Hmm add maybe a note for Condition or iterators?

Author: done

> The *Operators List* Plugin helps us visualize the list of operators in the workflow and shows the their current progress.

Reviewer: Should we change the name of this Plugin? We said via the chat that when we refer to the objects in the workflow we are talking about "Nodes", while when talking about the code / functionality we talk about "Operators". 
I think that given that the plugin refers to the nodes in the workflow, it should refer to them as nodes.

Reviewer: A card has been created to address this change. 
https://github.com/Inbibo/shift/issues/1103

![Opening a Plugin](/images/opening_a_plugin.gif)

> The *Variables Browser* allows users to parse the workflow variables currently set and check their content. This is especially helpful in large workflows where it is easy to lose track of the nodes previously executed.

Reviewer: We should add a link to the Variables page in variables.
Question, is it possible to delete variables with this plugin too? I'm not sure if that functionality was added at the end or not.

Author: yes it is possible with a right click context menu.
I just added a comment about it.

Reviewer: Has the link to the variables page been added? I don't see it in the paragraph

Author: Where should the "Variables Page" be? I only see a "Variable Node" page, are you referring to this one?

Reviewer: Yes

# Plugins

Shift supports the integration of custom Plugins. Shift Plugins are Qt based UIs which can interact with Shift to display additional information, perform actions or modify the behaviour of the application.

## Shift Native Plugins

Shift ships with two plugins which can be found in the *Tools* section in the top menu bar. The desired Plugin can be opened by clicking on the corresponding entry.

Once a Plugin is opened, the Shift UI can be customized with the best suited layout.

The *Operators List* Plugin helps us visualize the list of operators in the workflow and shows the their current progress.

![Opening a Plugin](/images/opening_a_plugin.gif)

> The *Variables Browser* allows users to parse the workflow variables currently set and check their content. This is especially helpful in large workflows where it is easy to lose track of the nodes previously executed. This Plugin also allows to delete specific workflow variables when developing your workflow. To do that right-click on the variable entry that should be deleted and select the *Delete* option.

Reviewer: ```suggestion
The *Variables Browser* allows users to parse the workflow variables currently set and check their content. This is especially helpful in large workflows where it is easy to lose track of which nodes were previously executed. This Plugin also allows to delete specific workflow variables set while working on a workflow. To do that `Right-click` on the variable entry to delete and select the *Delete* option.

```

# Plugins

Shift supports the integration of custom Plugins. Shift Plugins are Qt based UIs which can interact with Shift to display additional information, perform actions or modify the behaviour of the application.

## Shift Native Plugins

Shift ships with two plugins which can be found in the *Tools* section in the top menu bar. The desired Plugin can be opened by clicking on the corresponding entry.

Once a Plugin is opened, the Shift UI can be customized with the best suited layout.

> The *Operators List* Plugin helps us visualize the list of operators in the workflow and shows the their current progress.

Author: being uk based it makes sense! :)
done

> ![Opening a Plugin](/images/opening_a_plugin.gif)

Reviewer: Please, check how to add the html code from this card and apply it to this image: https://github.com/Inbibo/shift_docs/issues/45

Ensure it renders properly on the webpage.

Author: changed the link to html format and it works

> The *Variables Browser* allows users to parse the workflow variables currently set and check their content. This is especially helpful in large workflows where it is easy to lose track of which nodes were previously executed. This Plugin also allows to delete specific workflow variables set while working on a workflow. To do that `Right-click` on the variable entry to delete and select the *Delete* option.

Author: As the word Plugin is a bit too redundant  in the docs I removed it from the Node Outliner above so now they are aligned.

> ![Variables Browser Plugin](/images/variables_browser_plugin.gif)

Reviewer: Please, check how to add the html code from this card and apply it to this image: https://github.com/Inbibo/shift_docs/issues/45

Ensure it renders properly on the webpage.

Author: changed the link to html format and it works
